### ISSUE TITLE
confignetwork support configure a specified ethernet network interface

### ISSUE THREAD

**Issue by bybai** [xcat2/xcat-core#3426]
Feature request:
Currently, when `confignetwork/confignics` configure installnic, it find installnic ip from dhcp lease file, then configure this ip as installnic ip; During `switchdiscover --setup`, the dhcp lease contain one tmp ip for installnic, it may not be the final ip, but `confignetwork/confignics` will configure the tmp ip for the installnic, in this case, `confignetwork/confignics` lose the final ip. So `confignetwork/confignics` need have a way to handle this case. The background task is #3375

Now, `confignetwork` call `configeth` to configure ethernet ip for installnic and other nics.  configeth has the way to configure a specified ethernet network interface as follwoing,  `confignetwork` is exposed to user, so `confignetwork` can use `configeth` to realize the function.
`configeth` example:
```
[root@c910f05c01bc06 postscripts]# updatenode bybc0603 "configeth eth0 10.5.106.33 10_0_0_0-255_0_0_0"
bybc0603: xcatdsklspost: downloaded postscripts successfully
bybc0603: Mon Jul 10 22:56:20 EDT 2017 Running postscript: configeth
bybc0603: #XCAT_CONFIG
bybc0603: [I]: configeth on bybc0603: os type: debian
bybc0603: configeth on bybc0603: old configuration: 2: eth0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc pfifo_fast state UP group default qlen 1000
bybc0603:     link/ether 42:2b:ec:49:9c:b9 brd ff:ff:ff:ff:ff:ff
bybc0603:     inet 10.5.106.33/8 scope global eth0
bybc0603:        valid_lft forever preferred_lft forever
bybc0603: [I]: configeth on bybc0603: new configuration
bybc0603:        10.5.106.33, 10.0.0.0, 255.0.0.0, 10.0.0.103
bybc0603: 2: eth0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc pfifo_fast state UP mode DEFAULT group default qlen 1000
bybc0603: RTNETLINK answers: Cannot assign requested address
bybc0603: [I]: configeth on bybc0603: eth0 changed, modify the configuration files


bybc0603: postscript: configeth exited with code 0
bybc0603: Running of postscripts has completed.
[root@c910f05c01bc06 postscripts]#
[root@c910f05c01bc06 postscripts]#
[root@c910f05c01bc06 postscripts]# xdsh bybc0603 "cat /etc/network/interfaces.d/eth0"
bybc0603: auto eth0
bybc0603: iface eth0 inet static
bybc0603:   address 10.5.106.33
bybc0603:   netmask 255.0.0.0
bybc0603:   network 10.0.0.0
```

**Comment by daniceexi** [comment]
@bybai why the configntwork is run for a node which is running in genesis ? (discovery )

**Comment by bybai** [comment]
Hi @daniceexi , This feature is coming from `switchdiscover --setup`;  In `configonie --all` scripts,  it need change ip to static. Currently,  `configonie` use `$cmd = "updatenode $csw -P hardeths,enablesnmp,configinterface";` , we recommend use `confignetwork`, but `confignetwork` has issue. So I opened this feature request.  Here I mean `switchdiscover`.

**Comment by daniceexi** [comment]
@bybai get it, thanks.

**Comment by bybai** [comment]
Hi @cxhong , Could you review the pull request #3528? 
This pull request is for `configonie` to replace `hardeths`, could you have a try and give me more comments?

**Comment by bybai** [comment]
fixed in pull request #3528

**Comment by bybai** [comment]
From @cxhong , she verified this enhancements, I will closed it.

I verified this PR on one of cumulus switch
```
root@c685f2u28sw:/etc/network# cat interfaces
#XCAT_CONFIG
source /etc/network/interfaces.d/*
root@c685f2u28sw:/etc/network# cd interfaces.d/
root@c685f2u28sw:/etc/network/interfaces.d# cat eth0
auto eth0
iface eth0 inet static
  address 172.10.2.4
  netmask 255.255.0.0
  hwaddress ether 8c:ea:1b:e8:74:c0
  gateway 172.10.1.5
```
looks good. I will merge the code
